src/messages/xApp/XApp.tsx

```diff
@@ -0,0 +1,110 @@
+import { FC, useEffect, useState } from "react";
+import classnames from "classnames";
+
+import { useMessageContext } from "src/messages/hooks";
+import Typography from "src/common/Typography";
+import { PrimaryButton } from "src/common/Buttons";
+
+import { CloseIcon } from "src/assets/svg";
+import classes from "./xApp.module.css";
+import { IPluginXApp } from "@cognigy/socket-client/lib/interfaces/messageData";
+
+const XApp: FC = () => {
+	const { message, messageParams } = useMessageContext();
+
+	const shouldOpenImmediately =
+		!messageParams?.hasReply && (message?.data?._plugin as IPluginXApp)?.data?.immediateOpen;
+
+	const [showOverlay, setShowOverlay] = useState(shouldOpenImmediately);
+
+	const handleSubmit = (event: MessageEvent) => {
+		console.log("event", event);
+
+		if (sessionUrl.startsWith(event.origin) === false) {
+			return;
+		}
+
+		if (event.data.type !== "x-app-submit") {
+			return;
+		}
+
+		handleClose();
+	};
+
+	useEffect(() => {
+		function unsubscribe() {
+			window.removeEventListener("message", handleSubmit);
+		}
+
+		if (showOverlay) {
+			window.addEventListener("message", handleSubmit);
+		} else {
+			unsubscribe();
+		}
+
+		return () => {
+			unsubscribe();
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [showOverlay]);
```

Comment:
Whenever possible, it is always recommended to satisfy the hook dependencies.
In this case you can just include handleSubmit inside the useEffect and add sessionUrl in the dependencies, since is just a primitive value. I don't see any overhead on doing it, and we avoid warnings like that.

Reply:
handleSubmit is a closure, so it also needs to be wrapped in usecallback to stabilize its reference then. 
I changed as you requested, although I do not really like when we write code which only make sense for the linter